### PR TITLE
Divya

### PR DESCRIPTION
**Pull request by divyalakshmi-padmanaban**
1.create_elevator_array
2.input_floor
3.read_passenger
4.remove_passenger_floor
5.is_lift_on_floor

All the above functions are part of release 1

### CODE REVIEW

**Comment by Hariprasadm1408** [comment]
Looks fine to me

**Comment by ritika2** [comment]
Looks fine

**Comment by MaazJamal** [comment]
You have deleted files in devel from your branch. If we were to merge this those files would be deleted here as well.Add those files again to your branch and commit.

File: include/read_passenger.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+Enables to read the data from file and store in a structure of maximum capacity 100.
+
+The function  defination is found in read_passeneger.c 
+*/
+#ifndef __READ_PASSENGER_H__
+#define __READ_PASSENGER_H__
+#include "structures.h"
+
+extern int read passenger(struct passenger*p);
```
Comment (MaazJamal) [inline include/read_passenger.h]: You are missing an _ here. It should be read_passenger

File: src/create_elevator_array.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	struct elevator *ele;
+	ele=malloc (no_of_elevators * sizeof(struct elevator));//creates memory for the input-no_of_elevators.
+	if(ele==NULL)//checks if the memory allocation is success
+	{
+		printf("Memory allocation failed \n");
+		return 0;
+	}
+	else
+	{
+
```
Comment (MaazJamal) [inline src/create_elevator_array.c:71]: You also have to initialize the elevator array including initializing the passenger array inside the structure.

Comment (MaazJamal) [reply src/create_elevator_array.c:71]: Create a temp variable of type struct and than using a for loop over the array set all values of array equal to that value. Also in for loop create allocate memory for the passenger array and initialize there values as well. Set the in_elevator bool in passengers array to false.

File: src/read_passenger.c [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+
+**/
+void read_passenger(struct passenger p[])
+{
+
+    FILE *f;
+    int i=0;
+    int count=0;
+    f=fopen("C:\\bigfloor.txt","r");
```
Comment (MaazJamal) [inline src/read_passenger.c]: The path to text file should be local path. ("InputFiles/bigfloor.txt");

File: src/create_elevator_array.c (hunk 2 above)
Comment (divyalakshmi-padmanaban) [reply src/create_elevator_array.c:71]: Please let know what values to be assigned to array elements.In this function input parameter is just the number of elevators.

Comment (MaazJamal) [reply src/create_elevator_array.c:71]: bool	moving = 0.
bool	between_floor = 0
bool	direction_up = true
bool	direction_down = false
bool    *stop_at_floor = pointer to array of bool of length no_of_floors;
unsigned int	passenger_count = 0
unsigned int	max_passenger = 10
unsigned int	cur_floor = 0
unsigned int 	timer = 0
struct passenger *passenger_arr = pointer to array of size max passengers of type passenger.

**Review by Hariprasadm1408 — APPROVED** [review]

**Review by ritika2 — APPROVED** [review]

**Comment by MaazJamal** [comment]
> You have deleted files in devel from your branch. If we were to merge this those files would be deleted here as well.Add those files again to your branch and commit.

It fixed now.

**Pull request merged**